### PR TITLE
Simplify class metadata

### PR DESCRIPTION
**Pull request by annagrin**
Closes: https://github.com/dart-lang/webdev/issues/2101

### CODE REVIEW

File: dwds/lib/src/debugging/instance.dart [hunk 1]
```diff
@@ -868,60 +854,67 @@ class InstanceHelper extends Domain {
...
         return InstanceRef(
-          kind: InstanceKind.kPlainInstance,
+          kind: metaData.kind,
           id: objectId,
           identityHashCode: remoteObject.objectId.hashCode,
           classRef: metaData.classRef,
-        );
+        )..length = metaData.length;
+
+      //   if (metaData.isSystemList) {
```
Comment (annagrin) [inline dwds/lib/src/debugging/instance.dart:864]: remove:)

File: dwds/lib/src/debugging/metadata/class.dart [hunk 2]
```diff
@@ -69,18 +71,57 @@ LibraryRef libraryRefFor(String libraryId) => LibraryRef(
...
+  return ClassRef(
+    id: classIdFor(library, name),
+    name: name,
+    library: libraryRefFor(library),
+  );
+}
 
 String classIdFor(String libraryId, String? name) => 'classes|$libraryId|$name';
 
+enum ClassMetaDataKind {
```
Comment (annagrin) [inline dwds/lib/src/debugging/metadata/class.dart:86]: remove:)

**Comment by natebosch** [comment]
Is this change worth pursuing?

**Comment by annagrin** [comment]
> Is this change worth pursuing?

Not really, it is superseded by https://github.com/dart-lang/webdev/pull/2103

**Pull request closed without merging**